columnflow/production/scale.py

```diff
@@ -0,0 +1,96 @@
+# coding: utf-8
+
+"""
+Column production methods related to the renormalization and factorization scales.
+"""
+
+
+from columnflow.production import Producer, producer
+from columnflow.util import maybe_import
+from columnflow.columnar_util import set_ak_column
+
+ak = maybe_import("awkward")
+
+
+@producer(
+    uses={"LHEScaleWeight"},
+    produces={
+        "mur_weight", "mur_weight_up", "mur_weight_down",
+        "muf_weight", "muf_weight_up", "muf_weight_down",
+    },
+)
+def murmuf_weights(self: Producer, events: ak.Array, **kwargs) -> ak.Array:
+    """
+    Producer that reads out mur and muf uncertainties on an event-by-event basis.
+    Can only be called with MC datasets.
+
+    Resources:
+
+       - https://cms-nanoaod-integration.web.cern.ch/integration/master/mc94X_doc.html
+    """
+
+    # stop here for data
+    if self.dataset_inst.is_data:
+        raise ValueError("attempt to read out mur/muf weights in data")
+
+    N_scaleweights = ak.num(events.LHEScaleWeight, axis=1)
+    if ak.any(N_scaleweights != 9):
+        N_scaleweights = N_scaleweights[N_scaleweights != 9]
+        raise Exception(
+            f"Number of LHEScaleWeights ({N_scaleweights}) is not as expected (9) "
+            f"in dataset {self.dataset_inst.name}",
+        )
+
+    if ak.any(events.LHEScaleWeight[:, 4] != 1):
+        raise Exception("The nominal LHEScaleWeight should always be 1")
+
+    events = set_ak_column(events, "mur_weight", events.LHEScaleWeight[:, 4])
+    events = set_ak_column(events, "mur_weight_up", events.LHEScaleWeight[:, 7])
+    events = set_ak_column(events, "mur_weight_down", events.LHEScaleWeight[:, 1])
+    events = set_ak_column(events, "muf_weight", events.LHEScaleWeight[:, 4])
+    events = set_ak_column(events, "muf_weight_up", events.LHEScaleWeight[:, 5])
+    events = set_ak_column(events, "muf_weight_down", events.LHEScaleWeight[:, 3])
+
+    return events
+
+
+@producer(
+    uses={"LHEScaleWeight"},
+    produces={"scale_weight", "scale_weight_up", "scale_weight_down"},
```

Comment:
We should call the produced columns sth like `murf_envelope_weight{,_up,_down}`.